packages/kotti-ui/source/kotti-comment/KtComment.vue

```diff
@@ -95,75 +99,74 @@ export default {
 		KtCommentInput,
 	},
 	props: makeProps(KottiComment.propsSchema),
-	data() {
+	setup(props, { emit }) {
+		const isInlineEdit = ref(false)
+		const inlineMessageValue = ref<string | null>(null)
+		const userBeingRepliedTo = ref<UserData | null>(null)
+
+		const handleDelete = (commentId: number | string, isInline?: boolean) => {
+			const payload: KottiComment.Events.Delete = {
+				id: commentId,
+				parentId: isInline ? props.id : null,
+			}
+			emit('delete', payload)
+		}
 		return {
-			showInlineReply: false,
-			replyToText: null,
-			replyToUserId: Number,
-			isInlineEdit: false,
-			inlineMessageValue: '',
+			actionOptions: computed<Array<KottiButton.Props>>(() =>
+				compact([
```

Comment:
I really am not a fan of overuse of `lodash` utils just to remove `null`s here

this might have made more sense if we had falsible values because it apparently removes `''` and `false` and whatnot

So, I just either `.filter(Boolean)` at the end or not pushng to an array if it's not Editable
> you would have to declare the array as empty initially obviously and then conditionally push

some people have a third approach of pushing an empty array `[]`  in the false condition case  and then `flatMap`ping but I find this equally convoluted personally